core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -34,65 +35,54 @@ trait WorkspaceSupport {
       hasRequiredLevel =>
         if (hasRequiredLevel) {
-          val actionsBlockedByLock =
-            Set(SamWorkspaceActions.write, SamWorkspaceActions.compute, SamWorkspaceActions.delete)
-          if (actionsBlockedByLock.contains(requiredAction) && workspace.isLocked && !ignoreLock)
-            Future.failed(LockedWorkspaceException(workspace.toWorkspaceName))
-          else
-            Future.successful(())
+          Future.successful(())
         } else {
           samDAO.userHasAction(SamResourceTypeNames.workspace,
                                workspace.workspaceId,
                                SamWorkspaceActions.read,
                                ctx
           ) flatMap { canRead =>
-            if (canRead) {
-              Future.failed(WorkspaceAccessDeniedException(workspace.toWorkspaceName))
-            } else {
-              Future.failed(NoSuchWorkspaceException(workspace.toWorkspaceName))
-            }
+            if (canRead) Future.failed(WorkspaceAccessDeniedException(workspace.toWorkspaceName))
+            else Future.failed(NoSuchWorkspaceException(workspace.toWorkspaceName))
           }
         }
     }
 
-  def requireAccessF[T](workspace: Workspace, requiredAction: SamResourceAction)(codeBlock: => Future[T]): Future[T] =
-    accessCheck(workspace, requiredAction, ignoreLock = false) flatMap { _ => codeBlock }
+  def accessCheck(workspaceId: String, requiredAction: SamResourceAction): Future[Unit] =
+    samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, requiredAction, ctx) flatMap { hasRequiredLevel =>
+      if (hasRequiredLevel) {
+        Future.successful(())
+      } else if (requiredAction == SamWorkspaceActions.read) {
+        Future.failed(NoSuchWorkspaceException(workspaceId))
+      } else {
+        samDAO.userHasAction(SamResourceTypeNames.workspace, workspaceId, SamWorkspaceActions.read, ctx) flatMap {
+          canRead =>
+            if (canRead) Future.failed(WorkspaceAccessDeniedException(workspaceId))
+            else Future.failed(NoSuchWorkspaceException(workspaceId))
+        }
+      }
+    }
 
-  def requireAccessIgnoreLockF[T](workspace: Workspace, requiredAction: SamResourceAction)(
-    codeBlock: => Future[T]
-  ): Future[T] =
-    accessCheck(workspace, requiredAction, ignoreLock = true) flatMap { _ => codeBlock }
+  def checkLock(workspace: Workspace, requiredAction: SamResourceAction): Future[Unit] = {
+    val actionsBlockedByLock =
+      Set(SamWorkspaceActions.write, SamWorkspaceActions.compute, SamWorkspaceActions.delete)
+    if (actionsBlockedByLock.contains(requiredAction) && workspace.isLocked)
+      Future.failed(LockedWorkspaceException(workspace.toWorkspaceName))
+    else
+      Future.successful(())
+  }
 
   def requireComputePermission(workspaceName: WorkspaceName): Future[Unit] =
     for {
       _ <- userEnabledCheck
       workspace <- getWorkspaceContext(workspaceName)
-      workspaceId = workspace.workspaceId
```

Comment:
Now that the check for a workspace lock has been moved out of `accessCheck`, all this is now doing the same thing as `accessCheck`.